spec/unit/module_tool/applications/installer_spec.rb

```diff
@@ -247,6 +247,34 @@ def options
           expect(subject).to include :result => :success
           graph_should_include 'pmtacceptance-mysql', nil => v('0.8.0')
         end
+
+        context 'with an already installed dependency' do
+          let(:given_version) { '0.7.0' }
+
+          before do
+            preinstall('pmtacceptance-stdlib', '2.6.0')
+          end
+
+          def options
+            super.merge(:version => given_version)
+          end
+
+          it 'manages to install given module' do
+            expect(subject).to include :result => :success
+          end
+
+          it 'installs the given version' do
+            graph_should_include 'pmtacceptance-mysql', nil => v(given_version)
+          end
+
+          it 'does not change version of dependency' do
+            graph_should_include 'pmtacceptance-stdlib', v('2.6.0') => v('2.6.0')
+          end
+
+          it 'does not output any error' do
+            expect(subject[:error]).to be_nil
+          end
```

Comment:
What is the advantage of writing three partial tests versus above test where you have more checks in the same test?

Reply:
I thought it would make more sense to see a short description of every failed expectation when running the tests rather than a bundle of them with a long/generic description.